charmhelpers/contrib/database/mysql.py

```diff
@@ -547,36 +548,48 @@ def get_mem_total(self):
                     mtot, modifier = mem.strip().split(' ')
                     return '%s%s' % (mtot, modifier[0].upper())
 
-    def parse_config(self):
-        """Parse charm configuration and calculate values for config files."""
-        config = config_get()
-        mysql_config = {}
-        if 'max-connections' in config:
-            mysql_config['max_connections'] = config['max-connections']
+    def get_innodb_flush_log_at_trx_commit(self):
+        """Get value for innodb_flush_log_at_trx_commit.
 
-        if 'wait-timeout' in config:
-            mysql_config['wait_timeout'] = config['wait-timeout']
+        Use the innodb-flush-log-at-trx-commit or the tunning-level setting
+        translated by INNODB_FLUSH_CONFIG_VALUES to get the
+        innodb_flush_log_at_trx_commit value.
 
-        if 'innodb-flush-log-at-trx-commit' in config:
-            mysql_config['innodb_flush_log_at_trx_commit'] = \
-                config['innodb-flush-log-at-trx-commit']
-        elif 'tuning-level' in config:
-            mysql_config['innodb_flush_log_at_trx_commit'] = \
-                self.INNODB_FLUSH_CONFIG_VALUES.get(config['tuning-level'], 1)
+        :returns: Numeric value for innodb_flush_log_at_trx_commit
+        :rtype: int
+        """
+        if config_get('innodb-flush-log-at-trx-commit'):
+            return config_get('innodb-flush-log-at-trx-commit')
+        elif config_get('tuning-level'):
+            return self.INNODB_FLUSH_CONFIG_VALUES.get(
+                config_get('tuning-level'), 1)
 
-        if ('innodb-change-buffering' in config and
-                config['innodb-change-buffering'] in self.INNODB_VALID_BUFFERING_VALUES):
-            mysql_config['innodb_change_buffering'] = config['innodb-change-buffering']
+    def get_innodb_change_buffering(self):
+        """Get value for innodb_change_buffering.
 
-        if 'innodb-io-capacity' in config:
-            mysql_config['innodb_io_capacity'] = config['innodb-io-capacity']
+        Use the innodb-change-buffering validated against
+        INNODB_VALID_BUFFERING_VALUES to get the innodb_change_buffering value.
 
-        # Set a sane default key_buffer size
-        mysql_config['key_buffer'] = self.human_to_bytes('32M')
+        :returns: String value for innodb_change_buffering.
+        :rtype: str
+        """
+        _icb = config_get('innodb-change-buffering')
+        if _icb and _icb in self.INNODB_VALID_BUFFERING_VALUES:
+            return _icb
```

Comment:
Again, this is a partial function; it returns `None` if the if statement isn't True.  Either `:rtype: Optional[str]` or raise an `Exception` or return an empty `""`.

Reply:
Updating the docstring